### DISCUSSION TITLE
Docker CPU Allocation

### DISCUSSION THREAD

**Discussion by Finn0001** [itzg/docker-minecraft-server#1666] [Q&A]
Hello,

I'm looking at running a mod pack with around 300 mods. The server starts up very very slow and when I look in docker resource monitoring it's showing 150% CPU utilization. When I look at Task Manager it says 10% total. Does docker put some kind of CPU limit on containers?

**Comment by itzg** [comment]
Since you mention Task Manager, you're running with Docker Desktop right? If so, be aware that the containers are actually running within a VM that can be configured with a varying number of core allocation.

Take a look at https://docs.docker.com/desktop/settings/windows/

**Reply by itzg** [reply]
To answer your question another way, Docker doesn't impose CPU limits by default but if you chose to, you can explicitly apply various CPU limits if you look at the `--cpu*` type options in

https://docs.docker.com/engine/reference/commandline/run/#options

**Reply by Finn0001** [reply]
Hey, thanks for your quick response! I look into this solution, read through the documentation. I was unable to find the .wslconfig file under the "C:\Users\%UserProfile%" location. So I tried created my own file using that same file extension and it didn't seem to do anything.

The .wslconfig file I made contains:
[wsl2]
memory=30GB # Limits VM memory in WSL 2 to 4 GB
processors=20 # Makes the WSL 2 VM use two virtual processors

**Reply by itzg** [reply]
Great find...sounds like you're very close.
